CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -764,119 +781,107 @@ function Patient:tickDay()
+--! look for a drink.
+function Patient:_handleExcessThirst()
+  -- Callback function when the machine has been used
+  local --[[persistable:patient_drinks_machine_after_use]] function after_use()
+    self:changeAttribute("thirst", -(0.7 + math.random()*0.3))
+    self:changeAttribute("toilet_need", 0.05 + math.random()*0.05)
+    self:setMood("thirsty", "deactivate")
+    -- The patient might be kicked while buying a drink
+    if not self.going_home then
+      self.hospital:sellSodaToPatient(self)
     end
-    -- Don't check for a drinks machine too often
-    if self.timeout and self.timeout > 0 then
-      self.timeout = self.timeout - 1
-      return
+    -- The patient might also throw the can on the floor, bad patient!
+    if math.random() < 0.6 then
+      -- It will be dropped between 1 and 12 tiles away (litter bin catches 8 radius).
+      self.litter_countdown = math.random(1, 12)
     end
-    -- The only allowed situations to grab a soda is when queueing
-    -- or idling/walking in the corridors
-    -- Also make sure the walk action when leaving a room has a chance to finish.
-    if not self:getRoom() and not self:getCurrentAction().is_leaving and not self.going_home then
-      local machine, lx, ly = self.world:findObjectNear(self, "drinks_machine", 8)
-
-      -- If no machine can be found, resume previous action and wait a
-      -- while before trying again. To get a little randomness into the picture
-      -- it's not certain we go for it right now.
-      if not machine or not lx or not ly or math.random(1,10) < 3 then
-        self.timeout = math.random(2,4)
-        return
-      end
-
-      -- Callback function when the machine has been used
-      local --[[persistable:patient_drinks_machine_after_use]] function after_use()
-        self:changeAttribute("thirst", -(0.7 + math.random()*0.3))
-        self:changeAttribute("toilet_need", 0.05 + math.random()*0.05)
-        self:setMood("thirsty", "deactivate")
-        -- The patient might be kicked while buying a drink
-        if not self.going_home then
-          self.hospital:sellSodaToPatient(self)
-        end
-        -- The patient might also throw the can on the floor, bad patient!
-        if math.random() < 0.6 then
-          -- It will be dropped between 1 and 12 tiles away (litter bin catches 8 radius).
-          self.litter_countdown = math.random(1, 12)
-        end
-      end
+  end
 
-      -- If we are queueing, let the queue handle the situation.
-      for _, current_action in ipairs(self.action_queue) do
-        if current_action.name == "queue" then
-          local callbacks = current_action.queue.callbacks[self]
-          if callbacks then
-            callbacks:onGetSoda(self, machine, lx, ly, after_use)
-            return
-          end
+  -- Take the necessary steps to let the patient to get a soda.
+  local function constructGetSodaAction(machine, lx, ly)
```

Comment:
Why is this one not _constructGetSodaAction?

Reply:
It's already inside a private function. But I can rename it if you'd prefer.

Reply:
Yeah it is. It was only this way for readability but it can be dropped